docker/lwua-ingest/lwua-py/lwua/ingest.py

```diff
@@ -1,116 +1,49 @@
-from SPARQLWrapper import SPARQLWrapper, JSON
-from pathlib import Path
 import logging
-from rdflib import Graph
-import os
+import time
 from dotenv import load_dotenv
-from .helpers import enable_logging
+from SPARQLWrapper import SPARQLWrapper, JSON
+from pathlib import Path
+from .helpers import enable_logging,  data_path_from_config
+from .watcher import FolderChangeDetector, IngestChangeObserver
+from .graphdb import ingest_data_file, get_registry_of_lastmod
 
 
 log = logging.getLogger(__name__)
 
-
 def run_ingest():
     data_path = data_path_from_config()
     log.info(f"run_ingest on updated files in {data_path}")
-    # TODO -- immplement steps
-    # list all the contents (files) in data_path together with last mod
-    # get the <#admin-luwa-ingest> graph listing the maintained named-graphs and their lastmod
-    # there nees to be a mapping between filenames and named-graphs !
-    # check which filenames are younger then their named-graph equivalent
-    # read them into mem - replace the coresponding named-graph in the repo
-    # update the triple for the named-graph to lastmod in the admin grap
-
-
-def data_path_from_config():
-    folder_name = os.getenv("INGEST_DATA_FOLDER", "/data")
-    return Path(folder_name).absolute()
-
-
-def gdb_from_config():
-    base = os.getenv("GDB_BASE", "http://localhost:7200")
-    repoid = os.getenv("GDB_REPO", "lwua23")
-
-    endpoint = f"{ base }/repositories/{ repoid }"
-    updateEndpoint = endpoint + "/statements"   # update statements are handled at other endpoint
-
-    log.debug(f"using endpoint {endpoint}")
-
-    gdb = SPARQLWrapper(
-        endpoint=endpoint,
-        updateEndpoint=updateEndpoint,
-        returnFormat=JSON,
-        agent="lwua-python-sparql-client"
-    )
-    gdb.method = 'POST'
-    return gdb
-
-
-def ingest_graph(graph, gname: str = None, replace: bool = False):
-    log.debug(f"to insert data into <{ gname }>")
-
-    gdb = gdb_from_config()
+    
+    # get the last context graph modification dates
+    # run while true loop with 5 second sleep
+    detector = FolderChangeDetector(data_path)
+    ingestor = IngestChangeObserver()
+    last_mod = None
+    while last_mod is None:
```

Comment:
drop the while -- you only need to do this once
and let the scheduler do the retry